### PR TITLE
Compute on-road screen brightness based on camera exposure

### LINKED ISSUE #20705: Use exposure for screen brightness
Sometimes, a bad calibrated light sensor can get in the field. It would be much better if we use the camera exposure while onroad instead of the light sensor. Likely would fix things like https://github.com/commaai/openpilot/issues/20419 since the light sensor values vary from device to device.

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: selfdrive/ui/ui.cc [hunk 1]
```diff
@@ -177,8 +177,9 @@ static void update_state(UIState *s) {
   }
   if (sm.updated("sensorEvents")) {
     for (auto sensor : sm["sensorEvents"].getSensorEvents()) {
-      if (!Hardware::TICI() && sensor.which() == cereal::SensorEventData::LIGHT) {
-        scene.light_sensor = sensor.getLight();
+      // Use light sensor when offroad, camera exposure when onroad
+      if (!scene.started && Hardware::EON() && sensor.which() == cereal::SensorEventData::LIGHT) {
+        scene.light_sensor = std::clamp<float>(10.0 * sensor.getLight(), 0.0, 1023.0);
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc:182]: On older releases the screen brightness was max at a sensor value of `light_sensor = 255 / 2.6 = 98`. Currently we reach max brightness at `light_sensor = 100 / 0.1 = 1000`. By multiplying by 10, the old behavior is preserved.

File: selfdrive/ui/ui.cc [hunk 2]
```diff
@@ -319,7 +322,7 @@ void QUIState::update() {
 
 Device::Device(QObject *parent) : brightness_filter(BACKLIGHT_OFFROAD, BACKLIGHT_TS, BACKLIGHT_DT), QObject(parent) {
   brightness_b = Params(true).get<float>("BRIGHTNESS_B").value_or(10.0);
-  brightness_m = Params(true).get<float>("BRIGHTNESS_M").value_or(0.1);
+  brightness_m = Params(true).get<float>("BRIGHTNESS_M").value_or(2.6) / 26.0;
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc:325]: Old default was 2.6, and it was written to disk when the param was not present. We're stuk with this value.

File: selfdrive/ui/ui.cc [hunk 3]
```diff
@@ -193,10 +194,15 @@ static void update_state(UIState *s) {
       }
     }
   }
-  if (Hardware::TICI() && sm.updated("roadCameraState")) {
+  if (sm.updated("roadCameraState")) {
     auto camera_state = sm["roadCameraState"].getRoadCameraState();
-    float gain = camera_state.getGainFrac() * (camera_state.getGlobalGain() > 100 ? 2.5 : 1.0) / 10.0;
-    scene.light_sensor = std::clamp<float>((1023.0 / 1757.0) * (1757.0 - camera_state.getIntegLines()) * (1.0 - gain), 0.0, 1023.0);
+    if (Hardware::EON()) {
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui.cc:199]: should we check the inverse of the condition for the light sensor? you'll get weird behavior here with driver view for example

Comment (adeebshihadeh) [reply selfdrive/ui/ui.cc:199]: or better yet use different variables and handle this logic in the brightness code

File: selfdrive/ui/ui.cc [hunk 4]
```diff
@@ -193,10 +194,15 @@ static void update_state(UIState *s) {
...
+      float gain = camera_state.getGainFrac();
+      scene.light_sensor = std::clamp<float>((1023.0 / 5408.0) * (5408.0 - camera_state.getIntegLines()) * (1.0 - gain), 0.0, 1023.0);
```
Comment (ZwX1616) [inline selfdrive/ui/ui.cc:201]: This formula (max_t - t) * (max_gain - gain) feels a bit weird?

Comment (ZwX1616) [reply selfdrive/ui/ui.cc:201]: shouldn't it be
(max_t * max_gain) - (t * gain)
?

Comment (ZwX1616) [reply selfdrive/ui/ui.cc:201]: I think of it as the proportion of EV

**Comment by VirtuallyChris** [comment]
Is there any reason to keep using the light sensor while offroad? I've seen the consistency of the reading vary greatly depending on the device and lighting conditions. Can we just lock brightness to max while offroad?

File: selfdrive/ui/ui.cc (hunk 4 above)
Comment (pd0wm) [reply selfdrive/ui/ui.cc:201]: But if the gain goes up, wouldn't that influence the meaning of the max time as well? The current output seems to be pretty linear when I go through the whole range of exposure.

**Comment by ffhspa** [comment]
On my C2 at Night Screen its very bright, that disturb my Driving. Screen brightness should be lovered at Night

**Review by VirtuallyChris — APPROVED** [review]

**Pull request merged**